src/components/Button/Button.jsx

```diff
@@ -34,7 +34,18 @@ const buttonStyle = (variant, loading, modifier) => {
  * Generate the size class for this button, if applicable.
  * @see https://rivet.uits.iu.edu/components/forms/buttons/#small-buttons
  */
-const buttonSize = (size) => (size ? `${buttonClass}--${size}` : undefined);
+const buttonSize = (size) => {
+  if (!size) {
+    return undefined;
+  }
+
+  switch (size) {
+    case "full":
```

Comment:
If `small` and `full-width` are mutually exclusive why not use `full-width` as the property name so there's less business logic in here?